src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests.cs

```diff
@@ -125,9 +128,462 @@ public void M(int x)
+            let Number = int.Parse(STRING)
+            select Number * Number;
+    }
+}", new TestParameters(options: options.LocalNamesAreCamelCase));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestCamelCaseLocals_QueryLetClauseIgnored()
+        {
+            // This is an IRangeVariableSymbol, not ILocalSymbol
+            await TestMissingInRegularAndScriptAsync(
+@"using System.Linq;
+
+class C
+{
+    void M()
+    {
+        var squares =
+            from STRING in new string[] { }
+            let [|Number|] = int.Parse(STRING)
+            select Number * Number;
+    }
+}", new TestParameters(options: options.LocalNamesAreCamelCase));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestCamelCaseLocals_ParameterIgnored()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(int [|X|])
+    {
+    }
+}", new TestParameters(options: options.LocalNamesAreCamelCase));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestCamelCaseLocals_TupleTypeElementNameIgnored1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        (int [|A|], string B) tuple;
+    }
+}", new TestParameters(options: options.LocalNamesAreCamelCase));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestCamelCaseLocals_TupleTypeElementNameIgnored2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        (int A, (string [|B|], string C)) tuple = (0, (string.Empty, string.Empty));
+    }
+}", new TestParameters(options: options.LocalNamesAreCamelCase));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestCamelCaseLocals_TupleExpressionElementNameIgnored()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        var tuple = ([|A|]: 0, B: 0);
+    }
+}", new TestParameters(options: options.LocalNamesAreCamelCase));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestUpperCaseConstants_ConstField()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    const int [|field|] = 0;
+}",
+@"class C
+{
+    const int FIELD = 0;
+}",
+                options: options.ConstantsAreUpperCase);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestUpperCaseConstants_ConstLocal()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        const int local1 = 0, [|local2|] = 0;
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        const int local1 = 0, LOCAL2 = 0;
```

Comment:
❓ Where are local constants handled?

Reply:
I don't understand the question. Modifier checks are done by existing code on the symbols. I just added a test to confirm that it works for locals as well. So if you specify you want uppercase local constant, but normal locals otherwise, that should work and that's the purpose of this test.

Reply:
It is done here:
https://github.com/dotnet/roslyn/blob/70d99024f1bf802481ca5e378fca7fb840ee9ab0/src/Workspaces/Core/Portable/NamingStyles/Serialization/SymbolSpecification.cs#L397-L404

Reply:
tests added in 3f32ac5c839b806fdebebf6fb7ef19fb0399d7b6